include/pybind11/pytypes.h

```diff
@@ -353,6 +353,20 @@ class object : public handle {
         return *this;
     }
 
+#define PYBIND11_INPLACE_OP(inp)                                                                  \
```

Comment:
Very minor technically, but I'd definitely make this `iop` (a lot more intuitive).

Reply:
I actually think this is insufficient for attributes. Basically, we need to define the augmented operators for every object_api have custom operator=.

Reply:
Did you count already how many we have?
I'm not sure about the big picture TBH (I may still be misunderstanding), but this PR seemed like a necessary good step in the right direction, just by itself. Does that make sense?